### ISSUE TITLE
@internationalized/date ignores weeks when adding to ZonedDateTime

### ISSUE THREAD

**Issue by sanyabas** [adobe/react-spectrum#3667]
# 🐛 Bug Report 

Create `ZonedDateTime`, e.g. by calling `now` from `@internationalized/date` and add one week to it.
```ts
const d = now("Europe/Moscow");
console.log(d.toString()); //2022-10-20T10:12:00.797+03:00[Europe/Moscow]
console.log(d.add({ weeks: 1 }).toString()); //2022-10-20T10:12:00.797+03:00[Europe/Moscow] (!)
console.log(d.add({ months: 1 }).toString()); //2022-11-20T10:12:00.797+03:00[Europe/Moscow] works fine
```

btw CalendarDate works fine
```ts
const d = today("Europe/Moscow");
console.log(d.toString()); //2022-10-20
console.log(d.add({ weeks: 1 }).toString()); //2022-10-27
console.log(d.add({ months: 1 }).toString()); //2022-11-20
```

## 🤔 Expected Behavior

ZonedDateTime should work similar to CalendarDate

## 😯 Current Behavior

weeks are ignored

## 💁 Possible Solution

`addZoned` function checks only years, months and dates. Should it also include weeks?
https://github.com/adobe/react-spectrum/blob/bb28936482acb9ded909947a0cfc3f0e7fd785c1/packages/@internationalized/date/src/manipulation.ts#L367

## 💻 Code Sample

https://codesandbox.io/s/unruffled-meitner-kh03tb?file=/src/App.js:547-578

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| @internationalized/date   | 3.0.1

**Comment by jjasonlan** [comment]
looks like a missed case. would you be open to creating a PR for the fix?

**Comment by snowystinger** [comment]
Yep, looks like a missed check. Would definitely welcome a PR, thank you!

**Comment by pr7prashant** [comment]
I think this was fixed in #4702 and we can close this now.
